### ISSUE TITLE
utils.py的inner product实现有问题

### ISSUE THREAD

**Issue by CaoYiwei** [shenweichen/DeepMatch#93]
inner product是通过tf.reduce_sum实现的，reduce时候没有指定维度，建议默认axis=-1

当前0.3.1版本dssm模型无法构建，按如下方式修改后可以正常训练。

def inner_product(x, y, temperature=1.0, axis=-1):
    return Lambda(lambda x: tf.reduce_sum(tf.multiply(x[0], x[1]), axis) / temperature)([x, y])

**Comment by songcong123** [comment]
> inner product是通过tf.reduce_sum实现的，reduce时候没有指定维度，建议默认axis=-1
> 
> 当前0.3.1版本dssm模型无法构建，按如下方式修改后可以正常训练。
> 
> def inner_product(x, y, temperature=1.0, axis=-1): return Lambda(lambda x: tf.reduce_sum(tf.multiply(x[0], x[1]), axis) / temperature)([x, y])

遇到了同样的问题，我也感觉这一块有问题

**Comment by sosixyz** [comment]
是的，没有加axis,这样算的是一个value
